proxy/src/context.rs

```diff
@@ -0,0 +1,84 @@
+//! Connection request contexts
+
+use chrono::Utc;
+use once_cell::sync::OnceCell;
+use serde::{Serialize, Serializer};
+use smol_str::SmolStr;
+use std::net::IpAddr;
+use tokio::sync::mpsc;
+use uuid::Uuid;
+
+use crate::{error::ErrorKind, metrics::LatencyTimer};
+
+pub mod parquet;
+
+static LOG_CHAN: OnceCell<mpsc::UnboundedSender<RequestContext>> = OnceCell::new();
+
+#[derive(Clone, serde::Serialize)]
+/// Context data for a single request to connect to a database.
+///
+/// This data should **not** be used for connection logic, only for observability and limiting purposes.
```

Comment:
I was thinking about how to prevent someone from using it for connection logic. Otherwise I am afraid that it's just a matter of time until someone would use it for not-intended purposes.

What do you think about getting rid of public fields and adding logging/limitilg related methods instead. Than, I guess, name `RequestMonitoring` would fit it better.

Reply:
Sounds good to me